composer/callbacks/checkpoint_saver.py

```diff
@@ -456,12 +496,14 @@ def _save_checkpoint(self, state: State, logger: Logger):
                 src_path = str(pathlib.Path(saved_path).parent)
             else:
                 src_path = saved_path
-            this_rank_saves_symlinks = dist.get_global_rank() == 0 or not state.fsdp_sharded_state_dict_enabled
-            if this_rank_saves_symlinks:
+            rank_saves_symlinks = dist.get_global_rank() == 0 or not state.fsdp_sharded_state_dict_enabled
+            if rank_saves_symlinks:
                 os.symlink(os.path.relpath(src_path, os.path.dirname(symlink)), symlink)
 
         # if remote file name provided, upload the checkpoint
         if self.remote_file_name is not None:
+
+            futures: List[Future] = []
```

Comment:
```suggestion
            futures: list[Future] = []
```
I'm surprised this doesn't throw a typing error